macros/regex/regexp_instr.sql

```diff
@@ -1,33 +1,57 @@
-{% macro regexp_instr(source_value, regexp, position=1, occurrence=1, is_raw=False) %}
+{% macro regexp_instr(source_value, regexp, position=1, occurrence=1, is_raw=False, flags="") %}
 
     {{ adapter.dispatch('regexp_instr', 'dbt_expectations')(
-        source_value, regexp, position, occurrence, is_raw
+        source_value, regexp, position, occurrence, is_raw, flags
     ) }}
 
 {% endmacro %}
 
-{% macro default__regexp_instr(source_value, regexp, position, occurrence, is_raw) %}
+{% macro default__regexp_instr(source_value, regexp, position, occurrence, is_raw, flags) %}
+{# unclear if other databases support raw strings or flags #}
+{% if is_raw or flags %}
+    {{ exceptions.raise_compiler_error(
+            "is_raw and flags options not supported for this adapter"
+    ) }}
+{% endif %}
 regexp_instr({{ source_value }}, '{{ regexp }}', {{ position }}, {{ occurrence }})
 {% endmacro %}
 
 {# Snowflake uses $$...$$ to escape raw strings #}
-{% macro snowflake__regexp_instr(source_value, regexp, position, occurrence, is_raw) %}
+{% macro snowflake__regexp_instr(source_value, regexp, position, occurrence, is_raw, flags) %}
 {%- set regexp = "$$" ~ regexp ~ "$$" if is_raw else "'" ~ regexp ~ "'" -%}
-regexp_instr({{ source_value }}, {{ regexp }}, {{ position }}, {{ occurrence }})
+{% if flags %}{{ dbt_expectations._validate_flags(flags, 'cimes') }}{% endif %}
+regexp_instr({{ source_value }}, {{ regexp }}, {{ position }}, {{ occurrence }}, 0, '{{ flags }}')
 {% endmacro %}
 
 {# BigQuery uses "r" to escape raw strings #}
-{% macro bigquery__regexp_instr(source_value, regexp, position, occurrence, is_raw) %}
+{% macro bigquery__regexp_instr(source_value, regexp, position, occurrence, is_raw, flags) %}
+{% if flags %}
+    {{ exceptions.raise_compiler_error(
```

Comment:
We may need to make this a compiler warning?

Reply:
happy to downgrade that to a warning if you think that's better -- I just pushed a commit that should disable the two tests that use flags on BigQuery (those tests will fail on BQ if they don't raise).

Reply:
Looks like that still fails with the updated test https://app.circleci.com/pipelines/github/calogica/dbt-expectations/224/workflows/44eb5be9-8001-4c2e-a06c-fb6d39f31b72/jobs/136/parallel-runs/0/steps/0-106